src/include/btree_inline.h

```diff
@@ -743,8 +743,7 @@ __wt_page_only_modify_set(WT_SESSION_IMPL *session, WT_PAGE *page)
      * The page state can only ever be incremented above dirty by the number of concurrently running
      * threads, so the counter will never approach the point where it would wrap.
      */
-    if (__wt_atomic_load32(&page->modify->page_state) < WT_PAGE_DIRTY &&
-      __wt_atomic_add32(&page->modify->page_state, 1) == WT_PAGE_DIRTY_FIRST) {
+    if (__wt_atomic_add32(&page->modify->page_state, 1) == WT_PAGE_DIRTY_FIRST) {
```

Comment:
Good change! One atomic operation instead of two!

Reply:
It's not good - it used to be structured this way to avoid the write to the shared variable in most cases. That shared variable will be updated much more actively, but we haven't seen any performance consequences so far, so it seems as though it's going to be okay.